lib/xcvbnm.dart

```diff
@@ -1,33 +1,14 @@
 library xcvbnm;
 
-class Result {
-  // bits
-  double entropy;
-  // estimation of actual crack time, in seconds.
-  double crackTime;
-  // same crack time, as a friendlier string:
-  // "instant", "6 minutes", "centuries", etc.
-  String crackTimeDisplay;
-  // [0,1,2,3,4] if crack time is less than
-  // [10**2, 10**4, 10**6, 10**8, Infinity].
-  // (useful for implementing a strength bar.)
-  int score;
-  // how long it took xcvbnm to calculate an answer,
-  // in milliseconds.
-  int calcTime;
-}
-
-class Xcvbnm {
-  int getNaiveScore(String password) {
-    if (password.length < 3) return 0;
-    if (password.length < 5) return 1;
-    if (password.length < 7) return 2;
-    if (password.length < 9) return 3;
-    if (password.length < 11) return 3;
-    return 4;
-  }
+import 'src/xcvbnm_common.dart' hide Xcvbnm;
+export 'src/xcvbnm_common.dart' hide Xcvbnm;
+export 'src/xcvbnm_impl.dart';
+import 'src/xcvbnm_impl.dart';
 
-  Result estimate(String password, {List<String> userInputs}) {
-    return new Result()..score = getNaiveScore(password);
-  }
+/**
+ * the one and only entrypoint needed
+ */
+Result xcvbnm(String password, {List<String> userInputs}) {
```

Comment:
Exposing the function like this makes it redundant to actually have a `Xcvbnm` class. The whole point of the class is, that you can mock it out in your app.

Let's say, that you have a `register()` function in your app:

```dart
class RegistrationService {
  register(String username, String password) {
    var result = xcvbnm(password);
    if (result.score < 2) throw new Exception('Password not strong enough');
    _finishRegistratrion(username, password);
  }
}
```

There is no way to mock out xcvbnm and test that your library properly throws when xcvbnm returns a low score. Of course, you can provide a password where you assume that the score will be low, but then you are actually also testing xcvbnm... which is not a unit test.

If `Xcvbnm` is a class though, the function would be like this:

```dart
class RegistrationService {
  /// Can be replaced by a mock object
  var xcvbnm = new Xcvbnm();

  register(String username, String password) {
    var result = this.xcvbnm.estimate(password);
    if (result.score < 2) throw new Exception('Password not strong enough');
    _finishRegistratrion(username, password);
  }
}
```

Then you can mock it out in your test, and control the value returned:

```dart
test() {
  var service = new RegistrationService();
  var mockXcvbnm = new MockXcvbnm();

  // Overwrite the xcvbnm with the mock implementation
  service.xcvbnm = mockXcvbnm;

  // Configure the mock object with a controlled result
  when(mockXcvbnm.estimate(any)).thenReturn(new Result()..score = 0);
  
  // Now test that service.register(username, password); throws
}
```